mailapi/src/main/java/jakarta/mail/util/StreamProvider.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package jakarta.mail.util;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * Interface to be implemented by different mail implementations. It contains
+ * the methods to instance different encoders/decoders and other streams required by the API.
+ * 
+ * Implementations of this interface will need to be declared in
+ * META-INF/services/jakarta.mail.stream.StreamProvider
```

Comment:
this has to be made more general - sth like `Service lookup is used to find Implementations of this interface` (in the other words - current wording does not cover the case where the provider is defined in module-info)